aea/cli/search.py

```diff
@@ -103,121 +103,136 @@ def _get_details_from_dir(
         results.append(details)
 
 
+def _search_items(ctx, item_type_plural):
+    registry = cast(str, ctx.config.get("registry_directory"))
+    result = []  # type: List[Dict]
+    configs = {
+        "agents": {"loader": ctx.agent_loader, "config_file": DEFAULT_AEA_CONFIG_FILE},
+        "connections": {
+            "loader": ctx.connection_loader,
+            "config_file": DEFAULT_CONNECTION_CONFIG_FILE,
+        },
+        "contracts": {
+            "loader": ctx.contract_loader,
+            "config_file": DEFAULT_CONTRACT_CONFIG_FILE,
+        },
+        "protocols": {
+            "loader": ctx.protocol_loader,
+            "config_file": DEFAULT_PROTOCOL_CONFIG_FILE,
+        },
+        "skills": {
+            "loader": ctx.skill_loader,
+            "config_file": DEFAULT_SKILL_CONFIG_FILE,
+        },
+    }
+    if item_type_plural == "agents":
+        lookup_dir = registry
+    else:
+        lookup_dir = AEA_DIR
+        _get_details_from_dir(
+            configs[item_type_plural]["loader"],
+            registry,
+            "*/{}".format(item_type_plural),
+            configs[item_type_plural]["config_file"],
+            result,
+        )
```

Comment:
I'd change here to:
```suggestion
    if item_type_plural != "agents":
        _get_details_from_dir(
            configs[item_type_plural]["loader"],
            AEA_DIR,
            "*/{}".format(item_type_plural),
            configs[item_type_plural]["config_file"],
            result,
        )
```

Reply:
plus change in following comment make this bit easier to read

Reply:
Not a big deal, can change. I just prefer in cases where we have one condition and everything else goes under `else` to check that very condition in `if` statement. But I won't insist.